pydantic/deprecated/json.py

```diff
@@ -91,8 +96,12 @@ def pydantic_encoder(obj: Any) -> Any:
         raise TypeError(f"Object of type '{obj.__class__.__name__}' is not JSON serializable")
 
 
+@deprecated('custom_pydantic_encoder is deprecated, use BaseModel.model_dump instead.')
```

Comment:
```suggestion
@deprecated('custom_pydantic_encoder is deprecated')
```
I'm not sure what the replacement here is. @samuelcolvin is it possible to achieve something similar with `to_jsonable_python`?

Reply:
I see the `fallback` kwarg lets you handle the encoding of objects that would otherwise be non-serializable, but I don't see a way to override the encoding for specific types. @samuelcolvin it seems like it may be worth adding support for this in pydantic-core, which might also help resolve the xfailing `tests.test_forward_ref.test_json_encoder_str` (though would require support for something similar in `SchemaSerializer`).

In the meantime, my suggestion would be to also add a TODO note like this:
```suggestion
# TODO: Add a suggested migration path once there is a way to use custom encoders
@deprecated('custom_pydantic_encoder is deprecated.')
```
(happy to get Samuel's thoughts though.)